programs/zstdcli.c

```diff
@@ -544,6 +544,11 @@ static int init_cLevel(void) {
     return ZSTDCLI_CLEVEL_DEFAULT;
 }
 
+#define ZSTD_NB_STRATEGIES 9
```

Comment:
Find a way to compare this value with its source of truth.
Strategy indexes have bounds, which can be consulted with `ZSTD_cParam_getBounds()`.
Check that it corresponds.

Reply:
By check that it corresponds, you mean add an assert somewhere to make sure that it is within bounds?

Reply:
Yes.

Though "within bound" is another topic (mentioned below).

Here, the objective is merely to ensure that the table of names is relevant. In case the nb of strategies change in the future, we need a clear signal that this part needs also be updated.

Reply:
Actually I guess I could just use ZSTD_STRATEGY_MAX

Reply:
Yes, though : 
- It requires accessing the unstable section. We have successfully avoided that in `zstdcli.c` so far. In contrast, `ZSTD_cParam_getBounds()` is stable.
- It might still be worthy to use the constant instead of a function result _if_ the `assert()` was transformed into a _static assert_, in order to provide a compile-time signal. Unfortunately, no `static_assert()` macro is defined in `zstdcli.c`.